### PR TITLE
Add s2 cell union query.

### PR DESCRIPTION
**Pull request by feuerste**

### CODE REVIEW

File: src/s2_cells/mod.rs [hunk 1]
```diff
@@ -247,10 +250,14 @@ impl S2Cells {
             .collect();
         let mut cell_union = CellUnion(point_cells);
         cell_union.normalize();
-        let convex_hull = cell_union.rect_bound();
+        let rect = cell_union.rect_bound();
+        self.cells_intersecting(|cell: &Cell| rect.intersects_cell(cell))
+    }
+
+    fn cells_intersecting(&self, f: impl Fn(&Cell) -> bool) -> Vec<S2CellId> {
```
Comment (nnmm) [inline src/s2_cells/mod.rs]: Would it be a bit more elegant to just take a s2::Region?

Comment (feuerste) [reply src/s2_cells/mod.rs]: Good idea, I changed it.

File: src/s2_cells/mod.rs [hunk 2]
```diff
@@ -247,10 +250,14 @@ impl S2Cells {
...
         self.cells
             .values()
-            .filter(|cell| convex_hull.intersects_cell(cell))
+            .filter(|cell| f(cell))
```
[the remark this answers is not in the document]
Comment (feuerste) [reply src/s2_cells/mod.rs]: Not relevant any more after above change...

**Comment by feuerste** [comment]
Correct, there is no assumption about the cell union.

**Comment by feuerste** [comment]
Thanks for your review!

**Pull request merged**